### PR TITLE
Moving common logic of keyboard's simulation into the helper class.

### LINKED ISSUE #7506: Rework unit test infra for keyboard keypress simulations
### .NET version

- .NET 7

### Issue description

- Fix review points of #7489
- Needs to rework some unit tests to use a standard test helped of key press events

### PR DESCRIPTION
**Pull request by NikitaSemenovAkvelon**
Fixes #7506

## Proposed changes

- Added KeyboardHelper class in System.Windows.Forms.TestUtilities project.
- Used this class instead of copy-pasted code in unit tests.

## Customer Impact

- It's a technical refactoring.

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Unit tests

## Test environment(s) 

- 7.0.100-preview.5.22307.18

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7537)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
Moving to .NET8 because this is a refactoring.

File: src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using static Interop;
+using System.Windows.Forms;
+
+namespace System
+{
+    public static class KeyboardHelper
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs:6]: I would call this class `KeyboardSimulator` and remove the `simulate` prefix from all member names.

File: src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using static Interop;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs:1]: Please add the copyright header

File: src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        private static (nint keyCode, nint lParam) GetKeyParameters(Keys key)
+        {
+            // https://docs.microsoft.com/windows/win32/inputdev/wm-keyup
+            // The MSDN page tells us what bits of lParam to use for each of the parameters.
+            // All we need to do is some bit shifting to assemble lParam
+            // lParam = repeatCount | (scanCode << 16)
+            nint keyCode = (nint)key;
+            nint lParam = 0x00000001 | keyCode << 16;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/TestUtilities/KeyboardHelper.cs:28]: Please move appropriate method from Interop.PARAM.cs to build lparam

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, I added a few nits. Please don't merge until main becomes a NET8 branch

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Windows.Forms/tests/TestUtilities/KeyboardSimulator.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            User32.SendMessageW(control, User32.WM.KEYDOWN, keyCode, lParam);
+            User32.SendMessageW(control, User32.WM.KEYUP, keyCode, lParam);
+        }
+
+        private static (nint keyCode, nint lParam) GetKeyParameters(Keys key)
+        {
+            // https://docs.microsoft.com/windows/win32/inputdev/wm-keyup
+            // The MSDN page tells us what bits of lParam to use for each of the parameters.
+            // All we need to do is some bit shifting to assemble lParam
+            // lParam = repeatCount | (scanCode << 16)
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/TestUtilities/KeyboardSimulator.cs:30]: nit: this is different with the code below. So, I would just remove this line. We see `lParam` below.

**Review by vladimir-krestov — DISMISSED** [review]
Looks good! ✅

File: src/System.Windows.Forms/tests/TestUtilities/KeyboardSimulator.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            nint keyCode = (nint)key;
+            nint lParam = PARAM.ToInt(1, (int)key);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/TestUtilities/KeyboardSimulator.cs:32]: Or you could use more descriptive variable names and skip the comment completely:
```
int scanCode = (int)key;
int repeatCount = 1;
nint lParam=PARAM.FromLowHighUnsigned(low: repeatCount, high:scanCode);
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Please replace ToInt with FromLowHighUnsigned

**Review by RussKie — COMMENTED** [review]
🚀

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by vladimir-krestov — DISMISSED** [review]

**Comment by vladimir-krestov** [comment]
Please rebase to the latest main to resolve build errors

**Comment by Tanya-Solyanik** [comment]
>Please rebase to the latest main to resolve build errors

@vladimir-krestov - you should be able to re-run build by adding a `azp /run` comment to the PR. Could you please double-check if this works for you when you have a chance?

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by vladimir-krestov — APPROVED** [review]

**Comment by RussKie** [comment]
@Tanya-Solyanik this change only affects the test infra, is there any reason not to have this merged?

**Comment by Tanya-Solyanik** [comment]
>@Tanya-Solyanik this change only affects the test infra, is there any reason not to have this merged?

good point

**Pull request merged**